partition/roles/sonic/templates/frr.conf.j2

```diff
@@ -26,12 +26,21 @@ interface {{ port }}
  no ipv6 nd suppress-ra
 {% endfor %}
 {% for k, i in sonic_interconnects.items() %}
+{% if i.vrf is defined %}
+{% for n in i.unnumbered_interfaces|default([]) %}
+!
+interface {{ n }} vrf {{ i.vrf }}
```

Comment:
Why do we need `vrf` here? I never saw this before.

Reply:
With static machine ports we can have interconnects with `unnumbered_interfaces` in a vrf.
I changed the template to make what happens clearer; can you have another look?

Reply:
Interesting! :D

The documentation of frr I found is a bit vague, but it seems to suggest to use the `vrf` statement to add the interface to the vrf it belongs to.
Of course on Sonic there's also the `config_db.json` where this relation is also configured. I would suggest that it is safest to keep the Sonic and frr versions of the truth in sync.

This is also how the `metal-core` configures for regular machine ports; I wanted to replicate this behaviour closely so there is less of a chance for surprises.